common/script/libs/shops.js

```diff
@@ -0,0 +1,227 @@
+        value: foodItem.value,
+        locked: false,
+        currency: 'gems',
+        purchaseType: 'food',
+      };
+    }).sortBy('key').value();
+  categories.push(foodCategory);
+
+  return categories;
+};
+
+shops.getQuestShopCategories = function getQuestShopCategories (user, language) {
+  let categories = [];
+
+  _.each(content.userCanOwnQuestCategories, type => {
+    let category = {
+      identifier: type,
+      text: i18n.t(`${type}Quests`, language),
+    };
+
+    category.items = _(content.questsByLevel)
+      .filter(quest => quest.canBuy(user) && quest.category === type)
+      .map(quest => {
+        let locked = lockQuest(quest, user);
+        return {
+          key: quest.key,
+          text: quest.text(language),
+          notes: quest.notes(language),
+          value: quest.goldValue ? quest.goldValue : quest.value,
+          currency: quest.goldValue ? 'gold' : 'gems',
+          locked,
+          unlockCondition: quest.unlockCondition,
+          drop: quest.drop,
+          boss: quest.boss,
+          collect: quest.collect,
+          lvl: quest.lvl,
+          class: locked ? `inventory_quest_scroll_${quest.key}_locked` : `inventory_quest_scroll_${quest.key}`,
+          purchaseType: 'quests',
+        };
+      }).value();
+
+    categories.push(category);
+  });
+
+  return categories;
+};
+
+shops.getTimeTravelersCategories = function getTimeTravelersCategories (user, language) {
+  let categories = [];
+  let stable = {pets: 'Pet-', mounts: 'Mount_Head_'};
+  for (let type in stable) {
+    let category = {
+      identifier: type,
+      text: i18n.t(type, language),
+      items: [],
+    };
+
+    for (let key in content.timeTravelStable[type]) {
+      if (!user.items[type][key]) {
+        let item = {
+          key,
+          text: content.timeTravelStable[type][key](language),
+          class: stable[type] + key,
+          type,
+          purchaseType: type,
+          value: 1,
+          notes: '',
+          locked: false,
+          currency: 'hourglasses',
+        };
+        category.items.push(item);
+      }
+    }
+    if (category.items.length > 0) {
+      categories.push(category);
+    }
+  }
+
+  let sets = content.timeTravelerStore(user.items.gear.owned);
+  for (let setKey in  sets) {
+    let set = sets[setKey];
+    let category = {
+      identifier: set.key,
+      text: set.text(language),
+      purchaseAll: true,
+    };
+
+    category.items = _.map(set.items, item => {
+      return {
+        key: item.key,
+        text: item.text(language),
+        notes: item.notes(language),
+        type: item.type,
+        purchaseType: 'gear',
+        value: 1,
+        locked: false,
+        currency: 'hourglasses',
+        class: `shop_${item.key}`,
+      };
+    });
+
+    categories.push(category);
+  }
+
+  return categories;
+};
+
+shops.getSeasonalShopCategories = function getSeasonalShopCategories (user, language) {
+  let availableSets = {
```

Comment:
Is the idea to switch these out for whatever season we're in? Could you put a comment above this function to state that.